src/CMakeLists.txt

```diff
@@ -40,6 +40,9 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -rdynamic")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wstrict-null-sentinel -Woverloaded-virtual")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-new-ttp-matching")
+  if(NOT (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9))
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wpessimizing-move -Wredundant-move")
```

Comment:
i'd suggest test these flags before using it instead of relying on checking the compiler's version. as some of us might want to compile ceph using clang. and clang introduced these two flags in 3.7, see http://releases.llvm.org/3.7.0/tools/clang/docs/ReleaseNotes.html#id4 .

Reply:
done, thanks